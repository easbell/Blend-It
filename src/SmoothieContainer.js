import React, {Component} from 'react';
import Smoothie from './Smoothie.js';
import Recipe from './Recipe.js';

export default class SmoothieContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showRecipe: false,
      chosenSmoothie: ''
    }
  }

  showRecipe = (smoothie) => {
    const {showRecipe, chosenSmoothie} = this.state;
    this.setState({showRecipe: !showRecipe, chosenSmoothie: smoothie.id})
  }

  render = () => {
    const {showRecipe, chosenSmoothie} = this.state;
    let smoothieRecipe = this.props.smoothies.find(smoothie => {
      return smoothie.id === chosenSmoothie
    })
    return (
      <div>
        {!showRecipe && 
          this.props.smoothies.map(smoothie => {
            return (
              <Smoothie
              showRecipe={this.showRecipe}
              name={smoothie.name}
              img={smoothie.img}
              recipe={smoothie.recipe}
              key={smoothie.id}
              id={smoothie.id}
              />
              )
            })
          }
        {showRecipe && 
          <div>
            <Smoothie 
              showRecipe={this.showRecipe}
              name={smoothieRecipe.name}
              img={smoothieRecipe.img}
              recipe={smoothieRecipe.recipe}
              key={smoothieRecipe.id}
              id={smoothieRecipe.id}
            />
            <Recipe recipe={smoothieRecipe.recipe}/>
          </div>
        }
      </div>
    )
  }
}
