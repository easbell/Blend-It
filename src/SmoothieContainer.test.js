import React from 'react';
import SmoothieContainer from './SmoothieContainer.js';
import { shallow } from 'enzyme'

const mockSmoothies = [{
id: 1,
name: "Babe Ruth Strawberry Pineapple Banana Smoothie",
ingredients: [
"strawberries",
"pineapple",
"banana",
"orange juice",
"yogurt",
"spinach",
"chia seeds",
"flaxseed"
],
recipe: [
"1 cup strawberries",
"1/2 cup pineapple",
"1 banana",
"2 cups orange juice",
"1/2 cup greek yogurt",
"1 cup spinach, optional",
"1 tablespoon chia or flaxseed, optional",
"ice"
],
servings: 4,
source: "https://www.modernhoney.com/6-healthy-superfood-smoothies/",
img: "images/babe-ruth.jpg"
},
{
id: 2,
name: "Cherry Almond Smoothie",
ingredients: [
"milk",
"cherries",
"raw almonds",
"honey",
"cinnamon",
"almond extract",
"coconut oil"
],
recipe: [
"1 cup almond milk",
"1 cup cherries",
"3 tablespoons raw almonds",
"2 tablespoons raw honey",
"1 teaspoon cinnamon",
"1/2 teaspoon almond extract",
"1 tablespoon coconut oil"
],
servings: 1,
source: "https://bakedbree.com/cherry-almond-smoothies",
img: "images/cherry-almond.jpg"
}];


describe('SmoothieContainer', () => {
  const wrapper = shallow(
    <SmoothieContainer smoothies = {mockSmoothies} />
  );

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  

});







