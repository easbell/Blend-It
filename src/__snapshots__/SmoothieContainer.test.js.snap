// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmoothieContainer should match a snapshot with all data passed in 1`] = `
<div>
  <Smoothie
    id={1}
    img="images/babe-ruth.jpg"
    key="1"
    name="Babe Ruth Strawberry Pineapple Banana Smoothie"
    showRecipe={[Function]}
  />
  <Smoothie
    id={2}
    img="images/cherry-almond.jpg"
    key="2"
    name="Cherry Almond Smoothie"
    showRecipe={[Function]}
  />
</div>
`;
